### ISSUE TITLE
Inline property transitions

### ISSUE THREAD

**Issue by Dakror** [mikke89/RmlUi#448]
RCSS does not seem to support transitions for properties that are changed in the inline style. So for instance a progress bar that gets continuously updated in its width via the data model cannot be transitioned smoothly.

ElementStyle::TransitionPropertyChanges is only invoked for interpolation from old definition to new definition.

Real CSS supports this feature from what i've been able to test.

**Comment by mikke89** [comment]
Yeah, this is a known limitation of animations and transitions in the library. For simplicity, we essentially set the interpolated properties as local (inline) style. I believe in CSS, interpolation only affects the computed values.

I'm worried that adding another layer of properties (in addition to an element's definition and local style) will lead to bloat or worse performance. But maybe we can find a solution that works well for us.

**Comment by Dakror** [comment]
A somewhat dirty and probably poor-performing approach could be somehow modifying the definition through the data model or something, so that the current process gets used.

**Comment by mikke89** [comment]
I like the creativity, but this sounds too hacky for my taste ;) If we try to fix it, then it should also work when setting the style manually, any other behavior would be very surprising.
